### PR TITLE
chore(hotfix): separate message and displayMessage for FxError

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

File: packages/api/src/error.ts [hunk 1]
```diff
@@ -260,60 +186,16 @@ export class SystemError extends Error implements FxError {
...
     //other fields
     this.issueLink = option.issueLink;
     this.userData = option.userData;
-    this.notificationMessage = option.notificationMessage;
+    this.displayMessage = option.displayMessage;
     this.timestamp = new Date();
   }
 }
 
-/**
```
Comment (chagong) [inline packages/api/src/error.ts:277]: why remove it?

File: packages/api/src/error.ts [hunk 2]
```diff
@@ -336,7 +218,12 @@ export class UnknownError extends SystemError {
   }
 }
 
-export const UserCancelError: UserError = new UserError("UserCancel", "User canceled.", "UI");
+export const UserCancelError: UserError = new UserError(
+  "UI",
+  "UserCancel",
+  "User canceled.",
+  "User canceled."
```
Comment (chagong) [inline packages/api/src/error.ts:225]: can you use 'getLocalizedString' here?

File: packages/api/src/qm/visitor.ts [hunk 3]
```diff
@@ -219,10 +213,11 @@ const questionVisitor = async function (
...
   }
   return err(
-    returnUserError(
-      new Error(`Unsupported question node type:${JSON.stringify(question)}`),
+    new UserError(
       "API",
-      "UnsupportedNodeType"
+      "UnsupportedNodeType",
+      `Unsupported question node type:${JSON.stringify(question)}`,
+      `Unsupported question node type:${JSON.stringify(question)}`
```
Comment (chagong) [inline packages/api/src/qm/visitor.ts:220]: why not use 'getLocalizaedString'?

**Comment by jayzhang** [comment]
Change point: refactor constructors of SystemError and UserError, keeps only two constructors:
image
Separate two types of Error messages: message (for telemetry analsis) and displayMessage (for UI show) . 
We provide two functions for these two types of messages: getDefaultString() for message and getLocalizedString() for displayMessage.

**Review by LongOddCode — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by hund030 — APPROVED** [review]
Frontend and function part looks good to me.

**Review by kuojianlu — APPROVED** [review]

**Review by KennethBWSong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Review by meifans — APPROVED** [review]
LGTM on Deps Checker and ProjectMigrator

**Review by formulahendry — APPROVED** [review]
LGTM for Key Vault

**Review by xzf0587 — APPROVED** [review]
LGTM for identity and sql

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 4]
```diff
@@ -1479,14 +1520,22 @@ export async function wrapGetDeploymentError(
...
-    const returnError = new Error(
-      getLocalizedString(
-        "core.deployArmTemplates.FailedToGetDeploymentErrorNotification",
-        deployCtx.deploymentName,
-        deployCtx.resourceGroupName
+    return err(
+      new UserError(
+        SolutionSource,
+        "GetDeploymentErrorFailed",
+        getDefaultString(
```
Comment (wenytang-ms) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:1527]: why there need defaultString and localizedstring?

Comment (jayzhang) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:1527]: getDefaultString() is default English version used for telemetry.
getLocalizedString() is localized version used for UI display.

**Review by wenytang-ms — APPROVED** [review]
LGTM about ARM and migration

**Review by XiaofuHuang — APPROVED** [review]
LGTM for APIM

**Review by Alive-Fish — APPROVED** [review]

**Review by chagong — APPROVED** [review]

**Review by xiaolang124 — APPROVED** [review]

**Pull request closed without merging**